### PR TITLE
Add function to calculate CCA threshold from given dBm.

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/nrf_802154.c [hunk 1]
```diff
@@ -150,6 +150,11 @@ int8_t nrf_802154_dbm_from_energy_level_calculate(uint8_t energy_level)
     return ED_MIN_DBM + (energy_level / ED_RESULT_FACTOR);
 }
 
+uint8_t nrf_802154_ccaedthres_from_dbm_calculate(int8_t dbm)
```
Comment (KaSroka) [inline src/nrf_802154.c]: `ccaedthres` do we need all this information here and without underscores?

File: src/nrf_802154.h [hunk 2]
```diff
@@ -211,6 +211,16 @@ void nrf_802154_short_address_set(const uint8_t *p_short_address);
  */
 int8_t nrf_802154_dbm_from_energy_level_calculate(uint8_t energy_level);
 
+/**
+ * @brief  Calculate CCA energy detection threshold value from given dBm.
```
Comment (b-gent) [inline src/nrf_802154.h]: @brief  Calculate the CCA energy detection threshold value from a given dBm level.

@param[in]  dbm  Energy level in dBm used to calculate the CCAEDTHRES value.

@return  Energy level value corresponding to the given dBm level that should be written to the CCACTRL
register.

File: src/nrf_802154.c (hunk 1 above)
Comment (KaSroka) [reply src/nrf_802154.c]: Also I would omit `calculate` as `from_dbm` implies that it is somehow calculated from dBm.

Comment (hubertmis) [reply src/nrf_802154.c]: CCAEDTHRES is name of field in the register. We calculate value that should be placed in this register.
FCS suggests to use verb to end function name.

**Pull request merged**